packages/pigeon/lib/pigeon_lib.dart

```diff
@@ -2249,14 +2268,16 @@ ${_argParser.usage}''';
       options = options.merge(PigeonOptions(
           objcOptions: (options.objcOptions ?? const ObjcOptions()).merge(
               ObjcOptions(
-                  headerIncludePath: path.basename(options.objcHeaderOut!)))));
+                  headerIncludePath: options.objcOptions?.headerIncludePath ??
+                      path.basename(options.objcHeaderOut!)))));
```

Comment:
Is there a unit test that confirms this resolves https://github.com/flutter/flutter/issues/147587?

When I played with it when I discovered the issue, I also had to change a few other things: https://github.com/flutter/flutter/issues/147587#issuecomment-2086304905

Reply:
Still need to finish that part of the pr actually

Reply:
> Is there a unit test that confirms this resolves [flutter/flutter#147587](https://github.com/flutter/flutter/issues/147587)?


There is, but it bypasses the part of the code that was breaking this. 

I'll add one.